oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -1264,34 +1132,41 @@ private boolean loadItems() {
             try {
                 currentThread().setContextClassLoader(getClass().getClassLoader());
 
-                CloudBlobContainer container = azureBlobContainerProvider.getBlobContainer();
-                if (!firstCall && (resultContinuation == null || !resultContinuation.hasContinuation())) {
-                    LOG.trace("No more records in container. containerName={}", container);
+                if (!hasMorePages) {
+                    LOG.trace("No more records in container. containerName={}", containerClient.getBlobContainerName());
                     return false;
                 }
-                firstCall = false;
-                ResultSegment<ListBlobItem> results = container.listBlobsSegmented(null, false, EnumSet.noneOf(BlobListingDetails.class), null, resultContinuation, null, null);
-                resultContinuation = results.getContinuationToken();
-                for (ListBlobItem item : results.getResults()) {
-                    if (item instanceof CloudBlob) {
-                        items.add(AzureBlobInfo.fromCloudBlob((CloudBlob)item));
-                    }
+
```

Comment:
Is this needed? Can't we just return the blobItems.getIterator()? Pagination based on max results is already handled by the sdk, https://learn.microsoft.com/en-us/azure/developer/java/sdk/pagination

Reply:
Consider just using the sdk without any decorator as well. Just returning blobItems.getIterator() directly in the getAll* methods